File: Cpp_Trainings/towerHanoi.cpp
/*
CODEWARS
https://www.codewars.com/kata/5899f1df27926b7d000000eb

"Tower of Hanoi"
*/

#include <iostream>
#include <cmath>

long long towerOfHanoi(int rings)
{
    // Movimentos do Disco = (2^n) -1, n = número de discos.
    return pow(2, rings) - 1;
}

void testequal(long long ans, long long sol)
{
    std::cout << "ans) " << ans  << "\nsol) " << sol << std::endl;
    std::cout << std::boolalpha << (ans == sol) << std::endl;
}

int main()
{
    testequal(towerOfHanoi(4), 15);
    testequal(towerOfHanoi(5), 31);
    testequal(towerOfHanoi(10), 1023);

    return 0;  
};

File: Python_Trainings/towerHanoi.py
# -*- Mode: Python; coding: utf-8; indent-tabs-mpythoode: nil; tab-width: 4 -*-
# CODEWARS
# https://www.codewars.com/kata/5899f1df27926b7d000000eb
#
# "Tower of Hanoi"

def tower_of_hanoi(rings):
    # Movimentos do Disco = (2^n) -1, n = número de discos.
    return (2**rings) - 1

def testEqual(number1, number2):
    print(number1 == number2)

def main():
    print("Basic tests")
    testEqual(tower_of_hanoi(4),15)
    testEqual(tower_of_hanoi(5),31)
    testEqual(tower_of_hanoi(10),1023)
    testEqual(tower_of_hanoi(50),1125899906842623)

if __name__ == '__main__':
    main()